services/src/main/java/org/keycloak/theme/beans/LocaleBean.java

```diff
@@ -29,22 +29,31 @@
  */
 public class LocaleBean {
 
-    private String current;
+    private Locale current;
     private List<Locale> supported;
 
     public LocaleBean(RealmModel realm, java.util.Locale current, UriBuilder uriBuilder, Properties messages) {
-        this.current = messages.getProperty("locale_" + current.toLanguageTag(), current.toLanguageTag());
+        String currentTag = current.toLanguageTag();
```

Comment:
Just add field currentLanguageTag, rather than using Locale, as you don't need a new Locale object and generating the url for it.